Pull request for review

JudjinGM opened the pull request:
This is pull request for review my 8 spring homework

JudjinGM committed:
Made 4 screens (library and search is blank for now), implemented navigation between screens

JudjinGM committed:
Made 4 screens (library and search is blank for now), implemented navigation between screens

JudjinGM committed:
Merge remote-tracking branch 'origin/master'

JudjinGM committed:
Made 4 screens (library and search is blank for now), implemented navigation between screens

SeveNChaK commented:
К слову, не надо добавлять папку `.idea` к отслеживанию в `git`. Она у каждого своя и генерируется при создании/клонирование проекта на устройстве.

SeveNChaK commented on app/src/main/java/com/example/playlistmaker/MainActivity.kt:
```diff
@@ -1,11 +1,30 @@
+        settingButton.setOnClickListener {
+            val settingsIntent = Intent(this, SettingsActivity::class.java)
+            startActivity(settingsIntent)
+        }
```
⏫ Можно сократить чуток, сделав метод 
```
private fun navigateTo(clazz: Class<out AppCompatActivity>) {
    val intent = Intent(this, clazz)
    startActivity(intent)
}
```
и где нужно вызывать `navigateTo(SettingsActivity::class.java)` с соответствующим классом

JudjinGM replied:
Хорошо, спасибо!

SeveNChaK commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,98 @@
+
+    <TextView
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/content_fixed_height"
```
⏫ Лучше не использовать фиксированный размер для текстовых полей, потмоу что когда пользователь меняет размер текста в системе, то он увеличивается и в приложении и может не влезть в фиксирвоанный размер. Поэтому для текстовых полей лучше использовать относительные размеры `wrap_content` или `match_parent` или constrains, если используешь `ConstraintLayout`, а остальное задавать через отступы.

JudjinGM replied:
Спасибо, я поправлю!

SeveNChaK commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,98 @@
+            android:layout_height="wrap_content"
+            android:layout_marginStart="14dp"
+            android:layout_marginTop="14dp"
+            android:layout_marginBottom="61dp"
```
⚠️ Нпправильный отступ. Отсюда до свича должен быть отступ 40

JudjinGM replied:
Хорошо, спасибо за замечание, я поправлю!

JudjinGM committed:
Delete .idea directory

Удалил директорию .idea из отслеживания git

JudjinGM committed:
1. Добавил в gitignore директорию .idea
2. Добавил приватную функцию в MainActivity.kt для навигации
3. Исправил критическую ошибку в activity_settings.xml, исправил отступ до свитча
4. Убрал фиксированные размеры для текстовых полей в activity_settings.xml, заменил на отступы.

JudjinGM committed:
Merge remote-tracking branch 'origin/master'

JudjinGM committed:
1. Поправил файл .gitignore для директорий .idea
2. Испрваил критическую ошибку, неправильный отсуп в экране настройки, в файле activity_settings.xml. Поправил отступ до свича равный 40dp.
3. Добавил приватный метод в MainActivity.kt для навигации.
4. Убрал фиксированный размер текстовых полей, заменил на отсупы

JudjinGM committed:
1. Поправил файл .gitignore для директорий .idea
2. Испрваил критическую ошибку, неправильный отсуп в экране настройки, в файле activity_settings.xml. Поправил отступ до свича равный 40dp.
3. Добавил приватный метод в MainActivity.kt для навигации.
4. Убрал фиксированный размер текстовых полей, заменил на отсупы

JudjinGM committed:
1. Поправил файл .gitignore для директорий .idea
2. Испрваил критическую ошибку, неправильный отсуп в экране настройки, в файле activity_settings.xml. Убрал фиксированный размер текстовых полей, переделал все на constraintlayout.
3. Добавил приватный метод в MainActivity.kt для навигации.

JudjinGM committed:
1. Поправил файл .gitignore для директорий .idea
2. Испрваил критическую ошибку, неправильный отсуп в экране настройки, в файле activity_settings.xml. Убрал фиксированный размер текстовых полей, переделал все на constraintlayout.
3. Добавил приватный метод в MainActivity.kt для навигации.

JudjinGM committed:
1. Поправил файл .gitignore для директорий .idea
2. Испрваил критическую ошибку, неправильный отсуп в экране настройки, в файле activity_settings.xml. Убрал фиксированный размер текстовых полей, переделал все на constraintlayout.
3. Добавил приватный метод в MainActivity.kt для навигации.

JudjinGM committed:
Merge remote-tracking branch 'origin/master'

JudjinGM committed:
Merge remote-tracking branch 'origin/master'

JudjinGM committed:
Merge remote-tracking branch 'origin/master'
